.github/workflows/nix-linux.yml

```diff
@@ -10,18 +10,13 @@ permissions: read-all
 
 jobs:
   nix:
-    strategy:
-      fail-fast: false
-      matrix:
-        os: [ ubuntu-latest ]
-        mode: [ "flake check", "build" ]
-    runs-on: ${{ matrix.os }}
-    name: nix-build
+    runs-on: ubuntu-latest
+    name: nix build
```

Comment:
If you keep the `nix-build` name, we won't have to update the branch protection rules for this to be merged. If you prefer `nix build`, that's also fine, and I can update the rules.